lib/agent/grpc/request.go

```diff
@@ -148,26 +145,53 @@ func isRateLimitingThresholdExceeded(config *RateLimitingConfig, countsMap map[s
 	return counts.TotalNumberOfRequests >= config.MaxRequests
 }
 
-func getRateLimitingStatus(method string, route string, user string, ip string) *protos.RateLimitingStatus {
-	globals.RateLimitingMutex.RLock()
-	defer globals.RateLimitingMutex.RUnlock()
-
-	rateLimitingDataForRoute, exists := globals.RateLimitingMap[RateLimitingKey{Method: method, Route: route}]
+func getRateLimitingValue(method, route string) []*RateLimitingValue {
+	rateLimitingDataForEndpoint, exists := globals.RateLimitingMap[RateLimitingKey{Method: method, Route: route}]
 	if !exists {
-		return &protos.RateLimitingStatus{Block: false}
+		return []*RateLimitingValue{}
 	}
+	return []*RateLimitingValue{rateLimitingDataForEndpoint}
+}
 
-	if user != "" {
-		// If the user exists, we only try to rate limit by user
-		if isRateLimitingThresholdExceeded(&rateLimitingDataForRoute.Config, rateLimitingDataForRoute.UserCounts, user) {
-			log.Infof("Rate limited request for user %s - %s %s - %v", user, method, route, rateLimitingDataForRoute.UserCounts[user])
-			return &protos.RateLimitingStatus{Block: true, Trigger: "user"}
+func getWildcardRateLimitingValues(method, route string) []*RateLimitingValue {
+	wildcardRatelimitingValues := []*RateLimitingValue{}
+
+	for key, r := range globals.RateLimitingWildcardMap {
+		if key.Method != "*" && key.Method != method {
+			//If method is not '*', it must match exactly
+			continue
+		}
+		if r.RouteRegex.MatchString(route) {
+			wildcardRatelimitingValues = append(wildcardRatelimitingValues, r.RateLimitingValue)
 		}
-	} else {
-		// Otherwise, we rate limit by ip
-		if isRateLimitingThresholdExceeded(&rateLimitingDataForRoute.Config, rateLimitingDataForRoute.IpCounts, ip) {
-			log.Infof("Rate limited request for ip %s - %s %s - %v", ip, method, route, rateLimitingDataForRoute.IpCounts[ip])
-			return &protos.RateLimitingStatus{Block: true, Trigger: "ip"}
+	}
+	return wildcardRatelimitingValues
+}
+
+func getMatchingRateLimitingValues(method, route string) []*RateLimitingValue {
+	rateLimitingDataArray := getRateLimitingValue(method, route)
+	rateLimitingDataArray = append(rateLimitingDataArray, getRateLimitingValue("*", route)...)
+	rateLimitingDataArray = append(rateLimitingDataArray, getWildcardRateLimitingValues(method, route)...)
```

Comment:
You prioritize the defined method over "*" for non-wildcard routes but I don't see a sort for wildcard routes, i.e. even on wildcards a well-defined method is prioritized over a wildcard method

Reply:
In node : 
```ts
  possible.sort((a, b) => {
    if (a.method === b.method) {
      return 0;
    }

    if (a.method === "*") {
      return 1;
    }

    return -1;
  });
```